### PR TITLE
Fix: Make tools run on Linux and macos 

### LINKED ISSUE #242: tools: create_trustanchor.sh and create_callback_client_certificate.sh fail on Linux/macos
## Describe the bug

Running `./create_trustanchor.sh` or `./create_callback_client_certificate.sh` on any Linux or macos system fails (silently).

The created keystore is empty, because the generated X.509 certificate does not contain a subject (which was skipped). This seems to come from the usage of double forward-slashes and backward-slashes in the `openssl` command. `openssl` expects forward slashes for the `-subj` parameter. 

I can only assume that these double forward-slashes and backwards slashes were used to overcome [POSIX path conversion](http://www.mingw.org/wiki/Posix_path_conversion) issues of **MinGW/MSYS** which is used as part of **Git for Windows**.

## Expected behaviour

The bash script should create a keystore containing a certificate on POSIX systems.

## Steps to reproduce the issue

```
./create_trustanchor.sh
[1 of 5] Deleting old files...
rm: trustanchor.pem: No such file or directory
rm: efgs-ta.jks: No such file or directory
... old files deleted.
keystorefilename [efgs-ta.jks]:
keystorepassword [xyz]:
certCN [EFGS-TrustAnchor DEV]:
certC [DE]:
[2 of 5] Creating certificate...
Generating a 2048 bit RSA private key
............................+++
.....................+++
writing new private key to 'trustanchor.key'
-----
Subject Attribute /C has no known NID, skipped
... Certificate created.
[3 of 5] Creating keystore...
... keystore created.
[4 of 5] Adding certificate to keystore...
keytool error: java.lang.Exception: Input not an X.509 certificate
... certificate added.
[5 of 5] Deleting dummy cert from keystore...
... deleted dummy cert from keystore.
```

## Technical details

- Host Machine OS (Windows/Linux/Mac): Linux or macos

## Possible Fix

Use `-subj "/C=${certC}/CN=${certCN}/O=TrustAnchor Dev Org"` to create certificate.

### PR DESCRIPTION
**Pull request by UBaggeler**
This PR fixes #242 and #244 and improves the bash scripts to fail on error.

### CODE REVIEW

**Comment by UBaggeler** [comment]
Added 11581ef114effc459afb96aeb6501454d39f7cd2 to make generation of SQL INSERT more robust by using base64 to preserve newlines (instead of trying to escape them).

**Comment by daniel-eder** [comment]
Thank you for your contribution! Our devs will review your tickets and proposed fix and get back to you once capacity allows.

**Comment by ubamrein** [comment]
Hi there :)

Since this is now almost three months old, I was wondering what the state is?

File: tools/create_callback_client_certificate.sh [hunk 1]
```diff
@@ -28,8 +32,19 @@ certC=${input:-${certC}}
 #        * ) echo "Please answer Y(es) or N(o).";;
 # esac
 #done
+
+case "$(uname -s)" in
+  MINGW32*|MSYS*|MINGW*)
+    # Work around MinGW/MSYS's path conversion (http://www.mingw.org/wiki/Posix_path_conversion) 
+    ertSubject="//C=${certC}\CN=${certCN}\O=EFGS DEV Org"
```
Comment (f11h) [inline tools/create_callback_client_certificate.sh:39]:
```suggestion
    certSubject="//C=${certC}\CN=${certCN}\O=EFGS DEV Org"
```

typo

**Review by f11h — COMMENTED** [review]
Please review my recommended fix and then we can merge this PR

File: tools/create_callback_client_certificate.sh (hunk 1 above)
Comment (ubamrein) [reply tools/create_callback_client_certificate.sh:39]: Oh we missed that one! Thanks for the catch.

**Review by ubamrein — APPROVED** [review]

**Review by f11h — APPROVED** [review]

**Pull request merged**